const User = require("../models/user"); // importando o User que estÃ¡ dentro do models

const findById = async (req, res) => {
    console.log("ğŸš€ ~ findById ~ req:", req)
    try {
        const userId = req.params.id;
        console.log("ğŸš€ ~ findById ~ userId:", userId)
        const user = await User.findByPk(userId); //  Await - nÃ£o faz nada atÃ© retornar qualquer informaÃ§Ã£o desde erro atÃ© a informaÃ§Ã£o que procuro
        console.log("ğŸš€ ~ findById ~ user:", user)
        res.status(200).json(user); // Retorna o valor de user no formato json
    } catch (error) {
        res.status(500).json({status: 'error', message: 'Erro ao buscar o Usuario', error})
    }
    
} // async - consegue realizar requeciÃ§Ãµes ao mesmo tempo, req - vai ser todo o coropo da requisiÃ§Ã£o (solicitaÃ§Ã£o) (como por exemplo o id), res - response Ã© a resposta da sua aplicaÃ§Ã£o, mas nÃ£o estÃ¡ intricicamente ligado ao "req"

const createUser = async (req, res) => {
    console.log("ğŸš€ ~ createUser ~ req:", req)
    try {
        const userData = req.body;
        console.log("ğŸš€ ~ createUser ~ userData:", userData)
        const user = await User.create(userData); //  Await - nÃ£o faz nada atÃ© retornar qualquer informaÃ§Ã£o desde erro atÃ© a informaÃ§Ã£o que procuro
        console.log("ğŸš€ ~ createUser ~ user:", user)
        res.status(200).json(user); // Retorna o valor de user no formato json
    } catch (error) {
        res.status(500).json({status: 'error', message: 'Erro ao criar o Usuario', error})
    }
    
}


exports.userService = {
    findById, createUser
};
